app/controllers/api/v0x1/mixins/index_mixin.rb

```diff
@@ -0,0 +1,49 @@
+module Api
+  module V0x1
+    module Mixins
+      module IndexMixin
+        def list_order_items
+          collection(scoped(Order.find(params.require(:order_id)).order_items, OrderItem))
+        end
+
+        def list_orders
+          collection(scoped(Order.all, Order))
+        end
+
+        def list_portfolios
+          collection(scoped(Portfolio.all, Portfolio))
+        end
+
+        def list_portfolio_items
+          collection(scoped(PortfolioItem.all, PortfolioItem))
+        end
+
+        def fetch_portfolio_items_with_portfolio
+          collection(scoped(Portfolio.find(params.require(:portfolio_id)).portfolio_items, PortfolioItem))
+        end
+
+        def list_progress_messages
+          collection(scoped(OrderItem.find(params.require(:order_item_id)).progress_messages, ProgressMessage))
+        end
+
+        def scoped(relation, model)
+          if model.respond_to?(:taggable?) && model.taggable?
```

Comment:
@mkanoor I assume this comes directly from the Topology service since we have not done any tagging implementation to date.

Not sure if we keep this.  cc @syncrou

Reply:
@gmcculloug Yes its directly from topology

Reply:
Ladas has the tagging implementation in Topology (at least in a PR last week,)  so it's coming. But It appears the code nicely just uses relation if taggable doesn't exist.  If we go a different route we can remove it, but it seems we'll be using what they use anyway eventually.